### PR TITLE
Don't join multi-part literals

### LINKED ISSUE #391: [Suggestion] Don't join string literals that are split
Edit: (Related: https://github.com/solidity-parser/parser/issues/32 this is required to allow hex literals to have the same formatting)

I have a 148 character long string that is hard to look at in the source code. So I've split it using multi-line and prettier-ignore

Similar to JavaScript prettier, if I write a string as `'abc' + 'def'` than it tends to format things nicer than when I keep everything as one long string, the format process feels free to play around with my "chunks" and their position to make the "prettiest" format.

Currently split strings get joined.

Example:

```solidity
string memory a = 'abc'
                    'def'
          'ghi';
```

Current Result:

```solidity
string memory a = 'abcdefghi';
```

Suggestion:

```solidity
string memory a = 'abc'
    'def'
    'ghi';
```

Or maybe have them be same line unless going over a certain length:

```solidity
// assume including the ghi chunk would put it over the one line length limit.
string memory a = 'abcdef'
    'ghi';
// OR
string memory a = 'abc' 'def'
    'ghi';
```

### PR DESCRIPTION
**Pull request by fvictorio**
Closes #391

This PR prevents prettier from joining multi-part strings and hex literals. It also prints each one in a separate line when the line is too long.

### CODE REVIEW

File: tests/MultipartStrings/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    bytes b4 =
+        hex"beeeeeeeeeeeeeeeeeeeeeef"
+        hex"c0000000000ffeeeeeeeeeeeeeeeeeee";
+
+    string s1 = "foo";
+    string s2 = "foo" "bar";
+    string s3 =
+        "foofoofoofooofoofoofofoooofofoo"
+        "barbarbrabrbarbarbabrabrbabr";
+}
```
Comment (mattiaerre) [inline tests/MultipartStrings/__snapshots__/jsfmt.spec.js.snap:28]: very nice 👍

**Review by mattiaerre — APPROVED** [review]
👍

**Comment by junderw** [comment]
LGTM!!! Can't wait to clean up some contracts!

**Comment by junderw** [comment]
Also in semi-related issues to allow for multiline hex literals output by this new formatter:

https://github.com/sc-forks/solidity-coverage/issues/575

https://github.com/protofire/solhint/issues/273

**Comment by fvictorio** [comment]
@junderw they just need to update the parser to the latest version :)

**Pull request merged**